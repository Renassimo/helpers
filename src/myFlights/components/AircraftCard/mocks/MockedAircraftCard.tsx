const MockedAircraftCard = jest.fn(({ chosen }) => (
  <div>MockedAircraftCard - {chosen && 'chosen'}</div>
));

export default MockedAircraftCard;
