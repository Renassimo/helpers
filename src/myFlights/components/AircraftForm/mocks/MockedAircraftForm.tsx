const MockedAircraftForm = jest.fn(({ registration }) => (
  <div>MockedAircraftForm - {registration}</div>
));

export default MockedAircraftForm;
