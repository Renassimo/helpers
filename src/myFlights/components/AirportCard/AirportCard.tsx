import { Avia } from '@/avia/types/avia';

import AviaCard from '@/avia/components/AviaCard';

const AirportCard = ({
  data,
  onClick,
  chosen = false,
  title,
}: {
  data: Avia.AirportData;
  onClick: () => void;
  chosen?: boolean;
  title?: string;
}) => {
  const { attributes } = data;
  const { airportCode, airportName, municipalityName, shortName } = attributes;

  const withName =
    airportName ||
    `${municipalityName}${(municipalityName || shortName) && ' '}${shortName}`;

  const displayTitle = title ?? (airportCode || withName);

  const additionalContent = [
    title ? airportCode : null,
    displayTitle !== withName ? withName : null,
  ];

  return (
    <AviaCard
      onClick={onClick}
      chosen={chosen}
      title={displayTitle}
      additionalContent={additionalContent}
    />
  );
};

export default AirportCard;
