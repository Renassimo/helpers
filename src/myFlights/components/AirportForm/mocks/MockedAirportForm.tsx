const MockedAirportForm = jest.fn(({ title }) => (
  <div>MockedAirportForm - {title}</div>
));

export default MockedAirportForm;
