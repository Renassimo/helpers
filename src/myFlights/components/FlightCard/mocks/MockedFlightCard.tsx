const MockedFlightCard = jest.fn(({ chosen }) => (
  <div>MockedFlightCard - {chosen && 'chosen'}</div>
));

export default MockedFlightCard;
