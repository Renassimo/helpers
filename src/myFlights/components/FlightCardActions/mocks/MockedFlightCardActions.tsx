const MockedFlightCardActions = jest.fn(({ data }) => (
  <div>MockedFlightCardActions - {data?.id}</div>
));

export default MockedFlightCardActions;
