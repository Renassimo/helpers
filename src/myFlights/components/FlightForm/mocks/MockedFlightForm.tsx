const MockedFlightForm = jest.fn(() => <div>MockedFlightForm</div>);

export default MockedFlightForm;
