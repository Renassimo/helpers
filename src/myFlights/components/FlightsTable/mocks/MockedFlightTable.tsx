const MockedFlightTable = jest.fn(() => <div>MockedFlightTable</div>);

export default MockedFlightTable;
