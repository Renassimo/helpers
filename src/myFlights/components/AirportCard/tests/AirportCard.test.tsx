import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import AviaCard from '@/avia/components/AviaCard';

import MockedAviaCard from '@/avia/components/AviaCard/mocks';

import { mockedDeserializedAirports } from '@/avia/types/avia/mocks';

import AirportCard from '../AirportCard';

jest.mock('@/avia/components/AviaCard');

describe('AirportCard', () => {
  beforeEach(() => {
    (AviaCard as unknown as jest.Mock).mockImplementation(MockedAviaCard);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <AirportCard
        data={mockedDeserializedAirports[0]}
        chosen
        onClick={jest.fn()}
      />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when title passed', () => {
    test('renders successfully', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(
        <AirportCard
          data={mockedDeserializedAirports[0]}
          chosen
          onClick={jest.fn()}
          title="Origin"
        />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('when partial airport passed', () => {
    test('renders successfully', () => {
      // Arange
      const mockedAirport = {
        ...mockedDeserializedAirports[0],
        attributes: {
          ...mockedDeserializedAirports[0].attributes,
          airportName: null,
        },
      };
      // Act
      const { baseElement } = renderWithTheme(
        <AirportCard data={mockedAirport} onClick={jest.fn()} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });

    describe('and when title passed', () => {
      test('renders successfully', () => {
        // Arange
        const mockedAirport = {
          ...mockedDeserializedAirports[0],
          attributes: {
            ...mockedDeserializedAirports[0].attributes,
            airportName: null,
          },
        };
        // Act
        const { baseElement } = renderWithTheme(
          <AirportCard
            data={mockedAirport}
            title="Origin"
            onClick={jest.fn()}
          />
        );
        // Assert
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
