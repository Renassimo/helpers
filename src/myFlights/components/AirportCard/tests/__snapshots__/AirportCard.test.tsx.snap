// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirportCard renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedAviaCard - 
      WAW/EPWA
       -
      <div>
        Warsaw Chopin
      </div>
       
      - 
      chosen
       - 
       - 
       - 
    </div>
  </div>
</body>
`;

exports[`AirportCard when partial airport passed and when title passed renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedAviaCard - 
      Origin
       -
      <div>
        WAW/EPWA
      </div>
      <div>
        Warsaw Chopin
      </div>
       
      - 
       - 
       - 
       - 
    </div>
  </div>
</body>
`;

exports[`AirportCard when partial airport passed renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedAviaCard - 
      WAW/EPWA
       -
      <div>
        Warsaw Chopin
      </div>
       
      - 
       - 
       - 
       - 
    </div>
  </div>
</body>
`;

exports[`AirportCard when title passed renders successfully 1`] = `
<body>
  <div>
    <div>
      MockedAviaCard - 
      Origin
       -
      <div>
        WAW/EPWA
      </div>
      <div>
        Warsaw Chopin
      </div>
       
      - 
      chosen
       - 
       - 
       - 
    </div>
  </div>
</body>
`;
